### ISSUE TITLE
network behaves different in validating and testing?

### ISSUE THREAD

**Issue by MKimiSH** [torch/nn#994]
I am training a generative model. After several iterations of training I sample a batch from validation set for visualization. As the training goes on, the validating results get better and I saved the model for further use. But when I try to test it (with and without net:evaluate() because I forgot it at first) and the data is also from validation set, the result is so different from what I got before and is not acceptable. Also, I tried to finetune the model to check if it was saved wrongly, but it still looks good.
PS: The net contains several BatchNormalization modules. 
Is there anything else I missed when testing?

**Comment by MKimiSH** [comment]
Sorry I made a stupid mistake when loading input data... It works fine now...
